liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/InstallAppMojoSupport.java

```diff
@@ -216,11 +216,11 @@ private void addLibraryFromM2(Artifact library, LooseConfigData config) throws E
     
     private void validateAppConfig(String fileName) throws Exception {
         String appsDir = getAppsDirectory();
-        if (appsDir.equalsIgnoreCase("apps") && !isAppConfigInSourceServerXml()) {
+        if (appsDir.equalsIgnoreCase("apps") && !isFoundAppNameConfigInSourceServerXml(fileName)) {
             // add application configuration
             applicationXml.createWebApplicationElement(fileName);
         }
-        else if (appsDir.equalsIgnoreCase("dropins") && isAppConfigInSourceServerXml())
+        else if (appsDir.equalsIgnoreCase("dropins") && isFoundAppConfigInSourceServerXml())
```

Comment:
You only throw the exception if appsDirectory is `dropins` and the application `fileName` is configured